### ISSUE TITLE
[iOS] Gif Image tintColor not working

### ISSUE THREAD

**Issue by jinshin1013** [facebook/react-native#24789]
```
info
  React Native Environment Info:
    System:
      OS: macOS 10.14
      CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
      Memory: 22.33 MB / 16.00 GB
      Shell: 3.0.2 - /usr/local/bin/fish
    Binaries:
      Node: 10.15.3 - /usr/local/bin/node
      Yarn: 1.13.0 - /usr/local/bin/yarn
      npm: 6.4.1 - ~/.nvm/versions/node/v10.15.3/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.1, macOS 10.14, tvOS 12.1, watchOS 5.1
      Android SDK:
        API Levels: 23, 25, 26, 27, 28
        Build Tools: 26.0.2, 26.0.3, 27.0.3, 28.0.2, 28.0.3
        System Images: android-27 | Google APIs Intel x86 Atom, android-28 | Google APIs Intel x86 Atom, android-28 | Google Play Intel x86 Atom
    IDEs:
      Android Studio: 3.3 AI-182.5107.16.33.5314842
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.8.6 => 16.8.6
      react-native: 0.59.5 => 0.59.5
    npmGlobalPackages:
      react-native-cli: 2.0.1
```

## Steps To Reproduce

1. Create fresh react-native project through, `react-native init`
2. Create `Image source={require('path-to-image')} style={{ width: 50, height: 50, tintColor: "blue" }} />` 

When using an animated gif, `tintColor` style in Image is not being applied to iOS while working fine on Android. 

Snack: [Link to Expo Snack](https://snack.expo.io/HyA8drfhE)

**Comment by zhongwuzw** [comment]
@jinshin1013 Strange, can't reproducible, can you provide the image url? because tintColor only applied to non-alpha pixels.

**Comment by zhongwuzw** [comment]
Updated: based you snack code, it works for image snack-icon.png.
<img width="68" alt="WechatIMG55607" src="https://user-images.githubusercontent.com/5061845/57499055-ceefbe00-7310-11e9-832d-d26c4a0f4e84.png">

**Comment by jinshin1013** [comment]
@zhongwuzw Oops, my bad it seems like the issue is only contained in iOS gif images. I should've tested with the normal image as well before posting here, I just assumed image won't work either.

**Comment by jinshin1013** [comment]
Seems to be also related to #22805

**Comment by zhongwuzw** [comment]
@jinshin1013 Never mind, bug exist, I'll try to fix it. Thanks.

**Comment by zhongwuzw** [comment]
@jinshin1013 Hi, could you try #24794?

**Comment by jinshin1013** [comment]
@zhongwuzw I can confirm that the fix is working! Thanks!

**Comment by zhongwuzw** [comment]
@jinshin1013 🍻 

cc @cpojer @kelset. Fixed in #24794.
